frontend/src/lib/components/DateFilter/rollingDateRangeFilterLogic.ts

```diff
@@ -0,0 +1,117 @@
+import { actions, props, kea, listeners, path, reducers, selectors, defaults } from 'kea'
+import type { rollingDateRangeFilterLogicType } from './rollingDateRangeFilterLogicType'
+import { Dayjs } from 'lib/dayjs'
+import './RollingDateRangeFilter.scss'
+import { dateFilterToText } from 'lib/utils'
+
+const dateOptionsMap = {
+    q: 'quarter',
+    m: 'months',
+    w: 'weeks',
+    d: 'days',
+}
+
+export type RollingDateFilterLogicPropsType = {
+    selected?: boolean
+    onChange?: (fromDate: string) => void
+    dateFrom?: Dayjs | string | null
+}
+
+export const rollingDateRangeFilterLogic = kea<rollingDateRangeFilterLogicType>([
+    path(['lib', 'components', 'DateFilter', 'RollingDateRangeFilterLogic']),
+    actions({
+        increaseCounter: true,
+        decreaseCounter: true,
+        setCounter: (counter: number | null | undefined) => ({ counter }),
+        setDateOption: (option: string) => ({ option }),
+        toggleDateOptionsSelector: true,
+        select: true,
+    }),
+    props({} as RollingDateFilterLogicPropsType),
+    defaults(({ selectors }) => (_, props) => ({
+        dateOption: selectors.dateOptionDefault(_, props),
+        counter: selectors.counterDefault(_, props),
+    })),
+    reducers(() => ({
+        counter: [
+            3 as number | null | undefined,
+            {
+                increaseCounter: (state) => (state ? state + 1 : 1),
+                decreaseCounter: (state) => {
+                    if (state) {
+                        return state > 0 ? state - 1 : 0
+                    }
+                    return 0
+                },
+                setCounter: (_, { counter }) => counter ?? null,
+            },
+        ],
+        dateOption: [
+            'days',
+            {
+                setDateOption: (_, { option }) => option,
+            },
+        ],
+        isDateOptionsSelectorOpen: [
+            false,
```

Comment:
However, since all the selectors do is just convert `props` to a value, I'd personally either inline them as well, or convert to separate simple global functions that just take `(selected, dateFrom)` as arguments. E.g. `function counterDefault(selected, dateFrom) { return bla }`, and use them as inline defaults in the reducer.

Reply:
Done, thanks for the tip